src/main/java/org/assertj/core/error/ShouldEqualYear.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2022 the original author or authors.
+ */
+package org.assertj.core.error;
+
+import java.time.LocalDateTime;
+
+import org.assertj.core.internal.ComparisonStrategy;
+import org.assertj.core.internal.StandardComparisonStrategy;
+
+
+/**
+ * Creates an error message indicating that an assertion that verifies that a {@link LocalDateTime} is a given year failed.
+ *
+ * @author Jessica Hamilton
+ */
+public class ShouldEqualYear extends BasicErrorMessageFactory {
```

Comment:
To be consistent with date assertions, let's reuse `ShouldHaveDateField`, In `Dates.assertHasYear` it is used like `shouldHaveDateField(actual, "year", year)`:
```java
  public void assertHasYear(AssertionInfo info, Date actual, int year) {
    assertNotNull(info, actual);
    if (yearOf(actual) != year) throw failures.failure(info, shouldHaveDateField(actual, "year", year));
  }
```

We can remove `ShouldEqualYear` from this PR then.

Reply:
This makes sense, I removed it in a later commit.